tests/e2e/utils/prometheus/setup_prometheus_server.py

```diff
@@ -0,0 +1,172 @@
+                if key == line.split("=")[0]:
+                    line = f'{key}={original_to_replacement_dict.get(key)}\n'
+            updated_file_contents += line
+    with open(file_path, 'w') as updated_file:
+        updated_file.write(updated_file_contents)
+
+def update_params_env(workspace_id, region, params_env_file_path):
+    original_to_replacement_dict = {}
+    original_to_replacement_dict['workspaceId'] = workspace_id
+    original_to_replacement_dict['workspaceRegion'] = region
+    replace_params_env_in_line(params_env_file_path, original_to_replacement_dict)
+
+def create_AMP_workspace(region):
+    amp_client = boto3.client('amp', region_name=region)
+    
+    workspace_id = amp_client.create_workspace(alias = 'test_AMP_workspace')['workspaceId']
+    print(f"Created Workspace ID: {workspace_id}")
+
+    def check_active_workspace_status():
+        workspace_status = amp_client.describe_workspace(workspaceId=workspace_id).get("workspace").get("status").get("statusCode")
+        print(f"Workspace Status: {workspace_status}")
+        assert "ACTIVE" == workspace_status
+    
+    wait_for(check_active_workspace_status)
+    
+    return workspace_id
+
+def set_up_prometheus_for_AMP(cluster_name, region):    
+    associate_iam_oidc_provider(cluster_name, region)
+
+    policy_arn = create_AMP_ingest_policy()
+    
+    # Create AMP service account:
+    create_AMP_service_account_command = f'eksctl create iamserviceaccount --name {SERVICE_ACCOUNT_AMP_INGEST_NAME} --role-name {SERVICE_ACCOUNT_IAM_AMP_INGEST_ROLE} --namespace {PROMETHEUS_NAMESPACE} --cluster {cluster_name} --attach-policy-arn {policy_arn} --override-existing-serviceaccounts --approve --region {region}'.split()
+    subprocess.call(create_AMP_service_account_command, encoding="utf-8")
+
+    workspace_id = create_AMP_workspace(region)
+    
+    # Edit params.env to use workspace-id and region
+    update_params_env(workspace_id, region, f'{prometheus_yaml_files_directory}/params.env')
+    
+    create_namespace_command = f'kubectl create namespace {PROMETHEUS_NAMESPACE}'.split()
+    subprocess.call(create_namespace_command)
+    
+    kustomize_build_command = f'kustomize build {prometheus_yaml_files_directory}'.split()
+    print("About to run kustomize command.")
+    kustomize_build_output = subprocess.check_output(kustomize_build_command)
+    print("Finished running kustomize command.")
+    
+    print("About apply kustomize output.")
+    with tempfile.NamedTemporaryFile() as tmp_file:
+        tmp_file.write(kustomize_build_output)
+        tmp_file.flush()
+        kubectl_apply_command = f'kubectl apply -f {tmp_file.name}'.split()
+        subprocess.call(kubectl_apply_command, encoding="utf-8")
+    print("Finished applying kustomize output.")
+
+    return workspace_id
+
+def set_up_prometheus_port_forwarding():
+    print("Entered prometheus port_forwarding_function.")
+    def get_prometheus_pod_name():
```

Comment:
This method is named get_prometheus_pod_name but seems like a large portion of the code is focusing on describing the pod and making sure it has a certain status? Should this method be split up? If a pod has everything here but the `assert "1/1" == pod_status` fails then we are not able to retrieve the pod_name even though this is the prometheus pod is that intended?

Reply:
We can add a Todo for this since this method is only used once and the semantics are not particularly concerning.